File: 백준/Silver/18870. 좌표 압축/좌표 압축.py
import sys

n = int(input())

num_list = list(map(int, sys.stdin.readline().rstrip().split()))
dic_num = {}
result_list = []

num_set = set(num_list)         #중복제거
new_list = list(num_set)        #중복 제거된 숫자 리스트
new_list.sort()                 #정렬하기

for i in range(len(new_list)) : 
    dic_num[new_list[i]] = i    #딕셔너리에 {숫자 값 : 정렬된 순서} 저장                  

for i in range(len(num_list)) :      #num_list에 순서에 맞게 순서 result_list에 추가
    print(dic_num[num_list[i]], end=" ")

File: 백준/Silver/18870. 좌표 압축/좌표 압축.js
let fs = require("fs");
//let input = fs.readFileSync("./test.txt").toString().split("\n");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let n = Number(input[0]);
let arr = input[1].split(" ").map(Number);

//1. 중복된 값을 없애기 위해서 set 수행
const uniqueArr = [...new Set(arr)];

//2. 오름차순 정렬 수행
const sortArr = uniqueArr.sort((a, b) => a - b);
//3. 0부터 순서 매기기(순서가 자기보다 작은거 개수를 갖고있음)
let myMap = new Map();
for (let i = 0; i < uniqueArr.length; i++) {
    myMap.set(uniqueArr[i], i);
}

//4. Map 객체에 있는거 객체 순서 꺼내오기
let answer = "";
for (let item of arr) {
    //console.log(item, myMap.get(item));
    answer += myMap.get(item) + " ";
}

console.log(answer);
